app/api/chemotion/search_api.rb

```diff
@@ -97,10 +97,13 @@ def sample_structure_search(c_id = @c_id, not_permitted = @dl_s && @dl_s < 1)
 
         # TODO: implement this: http://pubs.acs.org/doi/abs/10.1021/ci600358f
         scope =
-          if params[:selection][:search_type] == 'similar'
+          case params[:selection][:search_type]
```

Comment:
This switch case seems to be replicating the switch case under `usecases/search/structure_search.rb`. Why?

Reply:
We discussed adding a fallback / default case for invalid `search_type`.

Reply:
usecases/search/structure_search.rb removed